src/main/java/org/broadinstitute/hellbender/tools/walkers/sv/PairedEndAndSplitReadEvidenceCollection.java

```diff
@@ -0,0 +1,434 @@
+            currentChrom = read.getContig();
+        } else {
+            flushSplitCounts(sp -> (sp.pos < readStart - 1), srWriter, splitCounts);
+        }
+
+        splitCounts.add(splitPosition);
+    }
+
+    private void flushSplitCounts(final Predicate<SplitPos> flushablePosition, final OutputStreamWriter srWriter, final PriorityQueue<SplitPos> splitCounts) {
+
+        while (splitCounts.size() > 0 && flushablePosition.test(splitCounts.peek())) {
+            SplitPos pos = splitCounts.poll();
+            int countAtPos = 1;
+            while (splitCounts.size() > 0 && splitCounts.peek().equals(pos)) {
+                countAtPos++;
+                splitCounts.poll();
+            }
+            try {
+                srWriter.write(currentChrom + "\t" + (pos.pos - 1) + "\t" + pos.direction.getDescription() + "\t" + countAtPos + "\t" + sampleName + "\n");
+            } catch (IOException e) {
+                throw new GATKException("Could not write to sr file", e);
+            }
+        }
+    }
+
+    private SplitPos getSplitPosition(GATKRead read) {
+        if (read.getCigar().getFirstCigarElement().getOperator() == CigarOperator.M) {
+            final int matchLength = read.getCigar().getCigarElements().stream().filter(e -> e.getOperator().consumesReferenceBases()).mapToInt(CigarElement::getLength).sum();
+            return new SplitPos(read.getStart() + matchLength, POSITION.RIGHT);
+        } else if (read.getCigar().getLastCigarElement().getOperator() == CigarOperator.M) {
+            return new SplitPos(read.getStart(), POSITION.LEFT);
+        }
+
+        return new SplitPos(-1, POSITION.MIDDLE);
+    }
+
+    private boolean isSoftClipped(final GATKRead read) {
+        final CigarOperator firstOperator = read.getCigar().getFirstCigarElement().getOperator();
+        final CigarOperator lastOperator = read.getCigar().getLastCigarElement().getOperator();
+        return (firstOperator == CigarOperator.SOFT_CLIP && lastOperator != CigarOperator.SOFT_CLIP) ||
+                (firstOperator != CigarOperator.SOFT_CLIP && lastOperator == CigarOperator.SOFT_CLIP);
+    }
+
+    @Override
+    public Object onTraversalSuccess() {
+        flushSplitCounts(splitPos -> true, srWriter, splitPosBuffer);
+        flushDiscordantReadPairs();
+        return null;
+    }
+
+    @Override
+    public void closeTool() {
+        super.closeTool();
+        try {
+            peWriter.close();
+            srWriter.close();
+        } catch (IOException e) {
+            throw new GATKException("error closing output file", e);
+        }
+    }
+
+    enum POSITION {
+        LEFT ("left"),
+        MIDDLE ("middle"),
+        RIGHT ("right");
+
+        private String description;
+
+        POSITION(final String description) {
+            this.description = description;
+        }
+
+        public String getDescription() {
+            return description;
+        }
```

Comment:
What's this for?

Reply:
Just to store the string that will actually make it into the file VS the name of the enum element. In this case it's just `"left"` vs `LEFT`, etc..